homeassistant/components/alexa/smart_home.py

```diff
@@ -927,12 +966,55 @@ def async_setup(hass, config):
     Even if that's disabled, the functionality in this module may still be used
     by the cloud component which will call async_handle_message directly.
     """
+
     smart_home_config = Config(
+        client_id=config[CONF_CLIENT_ID],
+        client_secret=config[CONF_CLIENT_SECRET],
         should_expose=config[CONF_FILTER],
         entity_config=config.get(CONF_ENTITY_CONFIG),
     )
     hass.http.register_view(SmartHomeView(smart_home_config))
 
+    global AUTH
+    AUTH = Auth(hass, smart_home_config.client_id,
+                smart_home_config.client_secret)
+
+    hass.loop.create_task(async_enable_proactive_mode(hass, smart_home_config))
+
+
+async def async_enable_proactive_mode(hass, smart_home_config):
+    if not await AUTH.async_is_auth_done():
+        # not ready yet
+        return
+
```

Comment:
we should add a check on the config to see if proactive mode is enabled.

Reply:
what do you mean?

Reply:
Not all users might want this functionality enabled because all of a sudden we're sharing a lot more information with Amazon. We should make it opt-in.

Reply:
If the user doesn't set the endpoint, client id and secret, the async_enable_proactive_mode() function will not be called (my latest commits improve on that).